### PR TITLE
Add system metrics to model-monitoring

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-7004
- [x] The Jira story is acked
- [ ] An entry has been added to the latest build document in [Build Announcements Folder](https://drive.google.com/drive/folders/1sgkK1WZgGo9CXsLizNe0GbAzVKuSKrZL).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious)

TESTING INSTRUCTIONS:
- Install rhods image on the cluster : `quay.io/vedantm/rhods-operator-liv
e-catalog:1.23.0-7004`
- Go to NS `redhat-ods-monitoring` -> routes -> open route for `rhods-model-monitoring`
- In the prometheus UI, verify if the metrics added in the PR show up as you start typing the metric name. eg: `container_cpu_usage_seconds_total`
- Verify all metrics added in this PR are available in the prometheus UI.

### CODE REVIEW

File: modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml [hunk 1]
```diff
@@ -11,6 +11,17 @@ spec:
         'match[]':
           - '{__name__= "haproxy_backend_http_average_response_latency_milliseconds"}'
           - '{__name__= "haproxy_backend_http_responses_total"}'
+          - '{__name__= "container_cpu_usage_seconds_total"}'
```
[the remark this answers is not in the document]
Comment (VedantMahabaleshwarkar) [reply modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml:14]: We're just enabling federating these metrics from openshift-monitoring right now. The queries for these metrics that we make (to show in the dashboard) will include the filter for just getting the metric for model serving pods

[the remark this answers is not in the document]
Comment (anishasthana) [reply modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml:14]: Right, my point is that we should only federate metrics we care about too.

**Review by lugi0 — CHANGES_REQUESTED** [review]
Apart from the dupe, everything looks good.
All queries autocomplete in Prometheus and correctly return data.

File: modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml [hunk 2]
```diff
@@ -11,6 +11,17 @@ spec:
...
+          - '{__name__= "container_memory_working_set_bytes"}'
+          - '{__name__= "node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate"}'
+          - '{__name__= "cluster:namespace:pod_cpu:active:kube_pod_container_resource_limits"}'
+          - '{__name__= "cluster:namespace:pod_cpu:active:kube_pod_container_resource_requests"}'
+          - '{__name__= "container_memory_working_set_bytes"}'
```
[the remark this answers is not in the document]
Comment (VedantMahabaleshwarkar) [reply modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml]: removed dupe

**Review by lugi0 — APPROVED** [review]
lgtm

File: modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (anishasthana) [reply modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml:14]: Can you comment in this thread with the JIRA link?

**Review by anishasthana — APPROVED** [review]

File: modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (VedantMahabaleshwarkar) [reply modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml:14]: https://issues.redhat.com/browse/RHODS-7020 is the jira for the follow up work

**Pull request merged**